src/content/docs/en/reference/experimental-flags/fonts.mdx

```diff
@@ -0,0 +1,682 @@
+
+
+## Font configuration reference
+
+All properties of your fonts must be configured in the Astro config. Some properties are common to both remote and local fonts, and other properties are available depending on your chosen font provider.
+
+### Common properties
+
+The following properties are available for remote and local fonts. `provider`, `name`, and `cssVariable` are required.
+
+```js title="astro.config.mjs"
+import { defineConfig } from "astro/config"
+
+export default defineConfig({
+    experimental: {
+        fonts: [{
+            provider: fontProviders.google(),
+            name: "Roboto",
+            cssVariable: "--font-roboto",
+            fallbacks: ["sans-serif"], // optional
+            automaticFallback: true, // default
+        }]
+    }
+})
+```
+
+#### provider
+
+<p>
+
+**Type:** `AstroFontProvider` | `"local"`
+</p>
+
+The source of your font files. You can use a [built-in provider](#available-remote-font-providers), write your own [custom provider](#custom-font-providers), or set to `"local"` to use local font files:
+
+```js title="astro.config.mjs"
+import { defineConfig, fontProviders } from "astro/config"
+
+export default defineConfig({
+    experimental: {
+        fonts: [{
+            provider: fontProviders.google(),
+        }]
+    }
+})
+```
+
+#### name
+
+<p>
+
+**Type:** `string`
+</p>
+
+The font family name, as identified by your font provider:
+
+```js title="astro.config.mjs"
+name: "Roboto"
+```
+
+#### cssVariable
+
+<p>
+
+**Type:** `string`
+</p>
+
+An identifier of your choice for your font, conventionally in the form of a CSS variable. It must be a [valid `<ident>`](https://developer.mozilla.org/en-US/docs/Web/CSS/ident):
+
+```js title="astro.config.mjs"
+cssVariable: "--font-roboto"
+```
+
+#### fallbacks
+
+<p>
+
+**Type:** `string[]`<br />
+**Default:** `["sans-serif"]`
+</p>
+
+An array of fonts to use when your chosen font is unavailable, or loading:
+
+```js title="astro.config.mjs"
+fallbacks: ["CustomFont", "serif"]
+```
+
+If the last fallback is a [generic family name](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#generic-name), Astro will try to generate an [optimized fallback](https://developer.chrome.com/blog/font-fallbacks) using font metrics.
+
+#### automaticFallback
+
+<p>
+
+**Type:** `boolean`<br />
+**Default:** `true`
+</p>
+
+Whether or not to enable automatic fallback font generation:
+
+```js title="astro.config.mjs"
+automaticFallback: false
+```
+
+### Remote font properties
+
+Further configuration options are available for remote fonts. Set these to customize the data loaded from your font provider, for example to only download certain font weights or styles.
+
+Under the hood, thse options are handled by [unifont](https://github.com/unjs/unifont/).
+
+#### weights
+
+<p>
+
+**Type:** `(number | string)[]`<br />
+**Default:** `[400]`
+</p>
+
+An array of [font weights](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight):
```

Comment:
By default only weight 400 for perf reasons. Be selective so you don't include too much fonts